lib/ansible/modules/windows/win_group_member.ps1

```diff
@@ -0,0 +1,230 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+function Test-GroupMember {
+    <#
+    .SYNOPSIS
+    Parse desired member into domain and username.
+    Also, ensure member can be resolved/exists on the target system by checking its SID.
+    .NOTES
+    Returns a hashtable of the same type as returned from Get-GroupMember.
+    Accepts username (users, groups) and domains in the following formats:
+    - username
+    - .\username
+    - SERVERNAME\username
+    - NT AUTHORITY\username
+    - DOMAIN\username
+    - username@DOMAIN
+    #>
+    param(
+        [String]$GroupMember
+    )
+
+    $parsed_member = @{
+        domain = $null
+        username = $null
+        combined = $null
+    }
+
+    # Split domain and account name into separate values
+    # '\' or '@' needs additional parsing, otherwise assume local computer
+
+    if ($GroupMember -match "\\") {
+        # DOMAIN\username
+        $split_member = $GroupMember.Split("\")
+
+        if ($split_member[0] -in @($env:COMPUTERNAME, ".")) {
+            # Local
+            $parsed_member.domain = $env:COMPUTERNAME
+        }
+        else {
+            # Domain or service (i.e. NT AUTHORITY)
+            $parsed_member.domain = $split_member[0]
+        }
+        $parsed_member.username = $split_member[1]
+    }
+    elseif ($GroupMember -match "@") {
+        # username@DOMAIN
+        $parsed_member.domain = $GroupMember.Split("@")[1]
+        $parsed_member.username = $GroupMember.Split("@")[0]
+    }
+    else {
+        # Local
+        $parsed_member.domain = $env:COMPUTERNAME
+        $parsed_member.username = $GroupMember
+    }
+
+    if ($parsed_member.domain -match "\.") {
+        # Assume FQDN was passed - change to NetBIOS/short name for later ADSI membership comparisons
+        $netbios_name = (Get-CimInstance -ClassName Win32_NTDomain -Filter "DnsForestName = '$($parsed_member.domain)'").DomainName
+
+        if (!$netbios_name) {
+            Fail-Json -obj $result -message "Could not resolve NetBIOS name for domain $($parsed_member.domain)"
+        }
+        $parsed_member.domain = $netbios_name
+    }
+
+    # Set SID check arguments, and 'combined' for later comparison and output reporting
+    if ($parsed_member.domain -eq $env:COMPUTERNAME) {
+        $sid_check_args = @($parsed_member.username)
+        $parsed_member.combined = "{0}" -f $parsed_member.username
```

Comment:
You might want to check this with a domain joined computer and validate the SID it returns when the local user also exists in the domain. I believe if you specify only the `username` argument to `NTAccount` and that username exists as both a local and domain user it will return the domain user. In this case it will return the SID for the domain user when in fact you want the local user. The trouble is that for local groups you can't set the `hostname` argument or it would fail.

I went a similar different route for determining the SID for a new module https://github.com/jborean93/ansible/blob/e3c728e3229b117501a1109dbeb050f3b680ee83/lib/ansible/modules/windows/win_user_right.ps1#L51-L113. I unfortunately don't have access to my test domain environment so can't 100% verify it.